rust/lance/src/dataset/scanner.rs

```diff
@@ -592,53 +620,101 @@ impl Scanner {
     /// 4. Limit / Offset
     /// 5. Take remaining columns / Projection
     async fn create_plan(&self) -> Result<Arc<dyn ExecutionPlan>> {
```

Comment:
This method is getting extremely complex. I hope we'll find the time soon to come up with a plan to refactor this.

Reply:
If we can get rid of some of these safeguards it will simplify things a little bit.  Also, maybe we can get rid of prefiltering / postfiltering if we can get the prefiltering performance to be good enough.

However, at the moment, there are hundreds of different possible plans that can be created...I don't have any good idea yet on how to refactor it simpler.

Reply:
Curouis how much we can push this to Datafusion's planner.